hdr_html_canvas_element.md

```diff
@@ -1,749 +1,313 @@
 
-The solution that we propose is to:
+* To render HDR imagery, it is useful to have information on the luminance range
+and color gamut that were used when authoring the image.
 
-* Add attributes to the `ScreenAdvanced` interface to expose screen parameters relevant to high dynamic range and wide color gamut.
-* Introduce new color spaces and precisions that are useful for HDR.
-* Clearly define invertible and context-independent transformations between these spaces.
-* Introduce the ability use more than 8 bits per pixel for a canvas element.
-* Introduce the ability for an `HTMLCanvasElement`'s rendering context to configure HDR metadata.
+Accordingly, the following API modifications are needed to manipulate HDR images
+in HTML Canvas:
 
-## Proposal
+1. add BT.2100 PQ and BT.2100 HLG color spaces to `PredefinedColorSpace`
+2. add higher bit depth capabilities to `CanvasRenderingContext2DSettings`
+3. add higher bit depth capabilities to `ImageDataSettings`
+4. add luminance and color gamut information to `ImageDataSettings` and
+   `CanvasRenderingContext2DSettings`
 
-### Querying screen high dynamic range parameters
+## Target use cases
 
-Add a new attribute to `ScreenAdvanced` to indicate the HDR headroom currently available on the screen.
+The primary use case is the drawing of HDR images into an HTML Canvas element
+such that the images are displayed as they would have been if they had been
+introduced in an `img` or `video` elements. Example applications include:
 
-```idl
-  partial interface ScreenAdvanced {
-    // The maximum luminance that the screen is capable of displaying across
-    // the full area of the screen, as a multiple of the luminance of SDR white.
-    // This will have a value of 1.0 for screens that are not HDR capable.
-    readonly attribute double highDynamicRangeHeadroom;
-  }
-```
-
-See below for more details on the display of floating point color values.
-
-### Querying screen wide color gamut parameters
-
-Add new attributes to `ScreenDetailed` to indicate the gamut and white point of the screen.
-
-```idl
-  partial interface ScreenDetailed {
-    // The color volume that the screen is capable of displaying.
-    readonly attribute ColorVolume colorVolume;
-  }
-```
-
-### Color volume structure
-
-Add a new color volume structure for use in several places.
-
-```idl
-  dictionary ColorVolume {
-    // The color primaries and white point of a color volume, in CIE 1931 xy
-    // coordinates. 
-    required double redPrimaryX;
-    required double redPrimaryY;
-    required double greenPrimaryX;
-    required double greenPrimaryY;
-    required double bluePrimaryX;
-    required double bluePrimaryY;
-    required double whitePointX;
-    required double whitePointY;
-  }
-```
+* drawing images retrieved from a file whose format is not supported by the
+  `img` or `video` elements
+* collages of images, both HDR and SDR, for a variety of sources
```

Comment:
I removed "variety of sources" since it was vague and unnecessary: it does not really matter where the pixels come from.

The idea is that the compositing happens in whatever colorspace the canvas is in, i.e. the incoming pixels are first converted to the Canvas' colorspace and then composited.